### DISCUSSION TITLE
Почему у поля ввода отсутствует выделение?

### DISCUSSION THREAD

**Discussion by omqmo** [taiga-family/taiga-ui#4617] [Q&A] [answered]
Если руками вызвать валидацию формы, то несмотря на отображение ошибки поле ввода не отображает обводку до наведения курсора.  Другими словами, поле ввода не реагирует на изменение валидатора реактивной формы angular.
Вот пример на [stackblitz](https://stackblitz.com/edit/stackblitz-starters-k6k8ts?description=An%20angular-cli%20project%20based%20on%20@angular/animations,%20@angular/common,%20@angular/compiler,%20@angular/core,%20@angular/forms,%20@angular/platform-browser,%20@angular/platform-browser-dynamic,%20@angular/router,%20core-js,%20rxjs,%20tslib%20and%20zone.js&file=src%2Fmain.ts&title=Angular%20Starter). Просто нажмите кнопку и все сами поймете. Если это не баг, то как это сделать рабочим?

**Comment by waterplea** [comment, marked as answer]
Компоненты в OnPush и недостаточно сделать markAsTouched, надо ещё updateValueAndValidity, чтобы выстрелило событие валидации.

**Reply by omqmo** [reply]
Вот так и знал, что это не может быть багом. Спасибо!
